### ISSUE TITLE
Chesapeake Label mask shape mismatch

### ISSUE THREAD

**Issue by recursix** [microsoft/torchgeo#515]
Hi all,

Some label mask have a shape of 512x512 instead of 256x256 when using ChesapeakeCVPRDataModule.

After investigation, I think they should be cropped to mask = mask[:256,:256]

**Comment by adamjstewart** [comment]
> I think they should be cropped to mask = mask[:256,:256]

I guess that depends on whether they cover the same extent as the images, in which case we would want to resample to the same resolution instead. Can you share the code needed to reproduce this issue? Like what bounding box can be used to reproduce this?

I think the best solution would be to inherit directly from RasterDataset instead of using a custom GeoDataset that seems to have bugs, but that will take a little longer to implement.

**Comment by recursix** [comment]
```python
    states = ["de", "md", "va", "wv", "pa", "ny"]

    dm = ChesapeakeCVPRDataModule(
        root_dir=SRC_DATASET_DIR,
        train_splits=[f"{state}-train" for state in states],
        val_splits=[f"{state}-val" for state in states],
        test_splits=[f"{state}-test" for state in states],
        patches_per_tile=500,
        patch_size=256,
        batch_size=1,
        num_workers=0,
        class_set=7,
    )

    dm.prepare_data()
    dm.setup()

    train_dl = dm.train_dataloader()
    val_dl = dm.val_dataloader()
    test_dl = dm.test_dataloader()

    n_samples = 0
    for s_idx, split_dl in enumerate([train_dl, val_dl, test_dl]):
        for i, dl_sample in enumerate(tqdm(split_dl)):
            sample_name = f"id_{n_samples:06d}"
            image = np.array(dl_sample["image"])[0]
            label = np.array(dl_sample["mask"])[0]
            crs = dl_sample["crs"][0]

            if label.shape != (256, 256):
                print(f"Cropping label to 256x256 from shape {label.shape}. ")
                label = label[:256, :256]
```

**Comment by recursix** [comment]
Hmmm, so far the reproducibility code didn't produce any (still 10h to go though).

**Comment by calebrob6** [comment]
The default transforms for the test dataloader pad 0s to `patch_size * 2`. Is this what you were observing?

**Comment by recursix** [comment]
Hmmm re-running everything made the problem disappear. I have no clue where it came from. This was a false alert. 

But hey... we got to chat a little bit ;)

**Comment by adamjstewart** [comment]
Glad it was just a false alarm, if you run into any other issues let us know!
